### ISSUE TITLE
Home Assistant Full Snapshot from 16 to 150 Mb

### ISSUE THREAD

**Issue by johanvh83** [jeroenterheerdt/ring-hassio#35]
Yesterday I implemented your repository into my Home Assistant.
Everything works fine. Every day I make a full snapshot.
Yesterday, before installation of your repository the full snapshot file was 16 Mb.
The snapshot today is 150 Mb!
Can you clarify this and eventually how to avoid this high disk usage

**Comment by DeliriousMetro** [comment]
This is as expected in HA, if you do full snapshot, this will include the media folder if you are using my automation of recording.

To prevent this from happening, perform partial snapshots instead in which you can deselect the media folder.

![media-unticked](https://i.imgur.com/I8e4ZkF.png)

If you want an automated approach of storing backups, I use the [Google Drive Hass.io Addon by sabeechen](https://github.com/sabeechen/hassio-google-drive-backup) and in the add-on configuration in the add-on page I exclude the media folder, so it will perform automatic partial snapshots to the cloud without the media folder included by adding `exclude_folders: media` in the add-on configuration tab.

**Comment by johanvh83** [comment]
I just made a full snapshot, 152,2 Mb and made partial snapshot without the Media folder.
The size was exactly the same and my Media folder is empty.
![Schermafbeelding 2021-01-21 om 16 54 36](https://user-images.githubusercontent.com/76775686/105375835-5d9d8f00-5c09-11eb-9200-13d1d942538e.png)

It has to do with the Add-on: Ring Livestream: 
When I disable this option the snapshot is only 16 Mb:
![Schermafbeelding 2021-01-21 om 16 53 39](https://user-images.githubusercontent.com/76775686/105375708-3ba40c80-5c09-11eb-9dc4-0ecf0e933b9a.png)

This add-on takes around 130 Mb! I suppose it contains all the Ring livestreams and will only grow in the future!
Where are the Livestreams stored by default? Once more: my Media folder is empty...

**Comment by DeliriousMetro** [comment]
Oh, I see, I was referring if you were saving videos with automation to the media folder because HA will backup that in a full snapshot.

I understand your issue now and can confirm it happens on my end as well and it is as expected lol.

When the addon is not running and you have not launched it for the first time, and you take a partial snapshot of this addon. The size is relatively small (0.1mb). This is because the addon has not been built yet.

You'll see that even though you ticked a partial snapshot of Ring Livestream, the addon is not present in your snapshot at all. Even in your referenced snapshot - it is missing when you disabled it.

![missing partial](https://i.imgur.com/ZtTXbcx.png)

Now you will compile the addon by pressing build and you see the large increase.

![partial-after](https://i.imgur.com/N5tCtl8.png)

When the addon is built and a partial snapshot of this addon is taken, the size is about 150mb as you say. I took a look inside the folder and it is due to the Node_modules which are built. This is designed on purpose as Node would not want to download the dependencies on the start of every runtime - that would be insane downloading 150mb to launch the addon lol. So it caches these in the modules folder.

These files are dependencies that is used to run the unofficial Ring API which this addon wraps around. Node_modules are usually downloaded from the web when Node.js launches but it will cache them in the 'Node_modules' folder so that these are not redownloaded again (saves you bandwidth).

**Comment by Corneloues** [comment]
I was interested in using this add-on, but 150Mb of dependent Node modules seems very inefficient. Are all these 100% necessary?

**Comment by tsightler** [comment]
It's difficult for me to see how he could cut this down, the dependency list for this project is already quite short.  The vast majority of the growth (~70MB), is from the dependency on ring-client-api, which you can't do without.  The reason ring-client-api is so big is because it depends on rxjs (~17MB), ffmpeg (~32MB), loadash (~5MB), and a few dozen other smaller dependencies.  There's nothing that this project could do about that.  I also maintain a project that uses ring-client-api and my addon is 145MB, so not much different.

Basically, 150MB is really nothing for a project of this type these days (as amazing as that sounds to a guy like myself that used to have to code assembler and fit it into a few KB).  I have several addons that are nearly 300MB in size!
